numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1192,145 @@ def infer_shapes(
+    sites. The amount of autocorrelation between sites is controlled by :math:`\alpha`. The
+    distribution is a popular prior for areal spatial data.
+
+    :param float or ndarray loc: mean of the multivariate normal
+    :param float alpha: autoregression parameter. For most cases, the value should lie
+        between 0 (sites are independent, collapses to an iid multivariate normal) and
+        1 (perfect autocorrelation between sites), but the specification allows for negative
+        correlations.
+    :param float tau: positive precision for the multivariate normal
+    :param numpy.ndarray or scipy.sparse.csr_matrix W: symmetric adjacency matrix where 1
+        indicates adjacency between sites and 0 otherwise
+    :param bool is_sparse: whether to use a sparse form of W in calculations (must be True is
+        W is a :class:`scipy.sparse.spmatrix`)
+    """
+
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.open_interval(-1, 1),
+        "tau": constraints.positive,
+        "W": constraints.dependent_property(is_discrete=False, event_dim=2),
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+        "W",
+    ]
+
+    def __init__(self, loc, alpha, tau, W, *, is_sparse=False, validate_args=None):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+
+        self.is_sparse = is_sparse
+
+        if self.is_sparse:
+            if not sparse.issparse(W):
+                assert isinstance(W, np.ndarray)
+            # TODO: look into future jax sparse csr functionality and other developments
+            self.W = sparse.csr_matrix(W)
+        else:
+            assert not sparse.issparse(W)
+            # TODO: look into static jax ndarray representation
+            assert isinstance(
+                W, np.ndarray
+            ), "only numpy arrays are currently supported"
+            self.W = W
+
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(loc)[:-1],
+            jnp.shape(alpha),
+            jnp.shape(tau),
+            jnp.shape(self.W)[:-2],
+        )
+        event_shape = jnp.shape(self.W)[-1:]
+        (self.loc,) = promote_shapes(loc, shape=batch_shape + event_shape)
+        self.alpha, self.tau = promote_shapes(alpha, tau, shape=batch_shape)
+
+        super(CAR, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+        if self._validate_args:
+            assert (
+                self.W.sum(axis=-1) > 0
+            ), "all sites in adjacency matrix W must have neighbours"
+
+            if sparse.issparse(self.W):
+                assert (self.W != self.W.T).nnz == 0, "W must be symmetric"
+            else:
+                assert np.array_equal(self.W, self.W.T), "W must be symmetric"
```

Comment:
Maybe changing `self.W.T` to `np.swapaxes(self.W, -2, -1)` to not limit ourselves to 2D W.

Reply:
I can't think of any occasions when W would not be 2D. Or what a multidimensional W would mean. It is a constant, a property of the sites' positions in space.

Reply:
I see. Then please just raise ValueError if W has batch dimensions.

Reply:
I've just used assert W.ndim == 2, but is there anything else to put?